examples/run/cpu/full_chain_algorithm.hpp

```diff
@@ -91,7 +92,7 @@ class full_chain_algorithm : public algorithm<track_state_container_types::host(
 
     private:
     /// Vecmem copy object
-    vecmem::copy m_copy;
+    std::shared_ptr<vecmem::copy> m_copy;
```

Comment:
Should this be `shared_ptr`?

Reply:
Using `std::shared_ptr` requires the least number of lines of change. 🤔

If I make it into an `std::unique_ptr` then I also need to provide a hand-written copy constructor and copy assignment operator. But with `std::shared_ptr` the auto-generated ones will do the correct thing.

Note that `vecmem::copy` has no internal state. So all its functions are by definition thread safe. Some of the device specific copy types are not thread safe, but the base / host class is.

With all this being said... Just relying on move construction/assignment may be enough... 🤔 I'll give that a try...